packages/blade/src/components/Card/__tests__/Card.native.test.tsx

```diff
@@ -0,0 +1,171 @@
+import { Text } from '~components/Typography';
+import { Counter } from '~components/Counter';
+import { Badge } from '~components/Badge';
+
+beforeAll(() => jest.spyOn(console, 'error').mockImplementation());
+afterAll(() => jest.restoreAllMocks());
+
+describe('<Card />', () => {
+  it('should render a plain Card', () => {
+    const { toJSON } = renderWithTheme(
+      <Card surfaceLevel={2}>
+        <CardBody>
+          <Text>Plain Card</Text>
+        </CardBody>
+      </Card>,
+    );
+    expect(toJSON()).toMatchSnapshot();
+  });
+
+  it('should render a Card with Header', () => {
+    const cardTitle = 'Card Header';
+    const cardSubtitle = 'Card subtitle';
+    const { getByText, toJSON } = renderWithTheme(
+      <Card surfaceLevel={2}>
+        <CardHeader>
+          <CardHeaderLeading
+            title={cardTitle}
+            subtitle={cardSubtitle}
+            prefix={<CardHeaderIcon icon={InfoIcon} />}
+            suffix={<CardHeaderCounter value={12} />}
+          />
+          <CardHeaderTrailing visual={<CardHeaderBadge>NEW</CardHeaderBadge>} />
+        </CardHeader>
+        <CardBody>
+          <Text>Plain Card</Text>
+        </CardBody>
+      </Card>,
+    );
+
+    expect(getByText(cardTitle)).toBeTruthy();
+    expect(getByText(cardSubtitle)).toBeTruthy();
+    expect(toJSON()).toMatchSnapshot();
+  });
+
+  it('should only accept allowed components in Card Header', () => {
+    const cardTitle = 'Card Header';
+    const cardSubtitle = 'Card subtitle';
+    expect(() =>
+      renderWithTheme(
+        <Card surfaceLevel={2}>
+          <CardHeader>
+            <CardHeaderLeading
+              title={cardTitle}
+              subtitle={cardSubtitle}
+              prefix={<InfoIcon color="action.icon.primary.default" size="xsmall" />}
+            />
+          </CardHeader>
+        </Card>,
+      ),
+    ).toThrow('[Blade CardHeaderLeading]: Only `CardHeaderIcon` component is accepted in prefix');
+
+    expect(() =>
+      renderWithTheme(
+        <Card surfaceLevel={2}>
+          <CardHeader>
+            <CardHeaderLeading
+              title={cardTitle}
+              subtitle={cardSubtitle}
+              suffix={<Counter value={12} />}
+            />
+          </CardHeader>
+        </Card>,
+      ),
+    ).toThrow(
+      '[Blade CardHeaderLeading]: Only `CardHeaderCounter` component is accepted in prefix',
+    );
+
+    expect(() =>
+      renderWithTheme(
+        <Card surfaceLevel={2}>
+          <CardHeader>
+            <CardHeaderTrailing visual={<Badge>NEW</Badge>} />
+          </CardHeader>
+        </Card>,
+      ),
+    ).toThrow(
+      '[Blade CardHeaderTrailing]: Only `CardHeaderLink, CardHeaderText, CardHeaderIconButton, CardHeaderBadge` component is accepted in visual',
+    );
+  });
+
+  it('should throw error if any sub card components are used outside of Card', () => {
+    expect(() => renderWithTheme(<CardHeader />)).toThrow(
```

Comment:
add test for CardBody being used outside Card?

Reply:
Done.